### ISSUE TITLE
wrong gradient descent for the ranking loss?

### ISSUE THREAD

**Issue by fqnchina** [princeton-vl/relative_depth#6]
Hi,

This is a great work. However, I'm a little confused of the loss function.

while r_k = +1, from what my understanding, the  gradient should be
-exp(-zi+zj)/(exp(-zi+zj)+1),
while in the implementation, it seems to be 
-/(exp(zi-zj)+1),

Correct me if I am wrong.

Many thanks!
